docs: Add note on resource address match order (#6818)
CloudFire-LLC/cloudfire: website/src/app/kb/deploy/resources/readme.mdx

  Resource.
</Alert>

#### Routing order for overlapping addresses

When multiple Resources' addresses overlap, the Resource with the more specific

> address will be used. For CIDR Resources, this means the Resource with the

Author: ```suggestion
address will be used.

For CIDR Resources, this means the Resource with the
```

> longest prefix length will be used.

Reviewer: Why is the "For CIDR Resources" part of this paragraph but for DNS, we start a new one?

Author: Hm I thought it was too choppy to split it up, but no strong feelings.

For DNS Resources, the more specific match will be used according to the
following rules:

> - Resources without wildcards are always prioritized over wildcard domains

Reviewer: ```suggestion
- Resources without wildcards are always prioritized over wildcard domains: For example, `app.example.com` is checked before `*.example.com`.
```

> - Single-char wildcards (`?`) take priority over label wildcards (`*`)

Reviewer: ```suggestion
- Single-char wildcards (`?`) take priority over label wildcards (`*`): For example, `???.example.com` is checked before `*.example.com`.
```

> - Label wildcards (`*`) take priority over catch-all wildcards (`**`)

Reviewer: ```suggestion
- Label wildcards (`*`) take priority over catch-all wildcards (`**`): For example, `*.example.com` is checked before `**.example.com`.
```

  Resource.
</Alert>

#### Routing order for overlapping addresses

When multiple Resources' addresses overlap, the Resource with the more specific
address will be used.

> For CIDR Resources, this means the Resource with the longest prefix length will

Reviewer: ```suggestion
For CIDR Resources, an address with a longer prefix is more specific than a shorter one. For example: `10.0.0.0/16` is more specific than `10.0.0.0/8`. IP Resources are essentially addresses with `/32` prefix and thus always more specific than any other CIDR.
```

This would avoid some word repetition.

be used.

> For DNS Resources, the more specific match will be used according to the

Reviewer: ```suggestion
For DNS Resources, more specific loosly translates to less wildcards. In particular:
```

  Resource.
</Alert>

#### Routing order for overlapping addresses

When multiple Resources' addresses overlap, the Resource with the more specific
address will be used.

For CIDR Resources, an address with a longer prefix is more specific than a
shorter one. For example: `10.0.0.0/16` is more specific than `10.0.0.0/8`. IP
Resources are essentially addresses with `/32` prefix and thus always more

> specific than any other CIDR. be used.

Reviewer: ```suggestion
specific than any other CIDR.
```

Couldn't make a multi-line comment when I made the original suggestion (was on my phone), so this one got lost.

> For DNS Resources, more specific loosly translates to less wildcards. In
> particular: following rules:

Reviewer: ```suggestion
For DNS Resources, more specific loosly translates to less wildcards. In
particular:
```
